### PR TITLE
Use sphinx-gallery for the doc

### PR DESCRIPTION
**Pull request by mehdidc**
- use the sphinx-gallery extension (https://github.com/sphinx-gallery/sphinx-gallery)
- make the examples compatible with sphinx-gallery 
- fix a dead link in xkcdify.py
- move examples/pyearth_vs_earth.py to benchmarks/

see there what would this look like : http://mehdidc.github.io/pyearth-doc/auto_examples/

### CODE REVIEW

File: examples/plot_classifier_comp.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
-'''
-This script recreates the scikit-learn classifier comparison example found at http://scikit-learn.org/dev/auto_examples/plot_classifier_comparison.html.
-It has been modified to include an Earth based classifier.
-'''
+"""
+===============================
+Plotting sckit-learn classifiers comparison with Earth
+===============================
```
Comment (agramfort) [inline examples/plot_classifier_comp.py]: number of = is not enough. I guess it breaks rendering. It should be the length of the number of characters in headline.

File: examples/plot_derivatives.py [hunk 2]
```diff
@@ -1,6 +1,15 @@
+"""
+===============================
+Plotting derivatives of simple sine function
+===============================
```
Comment (agramfort) [inline examples/plot_derivatives.py]: same rmk here.

File: examples/plot_classifier_comp.py [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
 
+This script recreates the scikit-learn classifier comparison example found at http://scikit-learn.org/stable/auto_examples/classification/plot_classifier_comparison.html.
+It has been modified to include an Earth based classifier.
+"""
```
Comment (agramfort) [inline examples/plot_classifier_comp.py]: I suspect you the `print(__doc__)`

**Comment by agramfort** [comment]
besides LGTM.

looking forward v0.1 release !

**Comment by jcrudy** [comment]
Thanks. Currently traveling without computer, but will be back around the 15th.

**Comment by mehdidc** [comment]
Thanks @agramfort for the remarks, fixed.

**Comment by jcrudy** [comment]
Just merged.  Should sphinx and the necessary extensions (sphinx-gallery, sphinx_rtd_theme, numpydoc) be listed as dependencies somehow in setup.py?  It's not clear to me how documentation dependencies are supposed to work.

**Comment by mehdidc** [comment]
Yes ! "sphinx-gallery" should be added  to "install_requires" in setup_args (in setup.py), I will add that.

**Pull request merged**